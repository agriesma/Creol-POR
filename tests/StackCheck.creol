class StackCheck
begin
  var depth : Int
  var max : Int
  op init == depth := 0; max := 3
  op run == test(;)
  op test ==
    var temp: Int; depth := depth + 1; temp := depth;
    if depth <= max then test(;) end;
    await depth = temp; depth := depth - 1
end
