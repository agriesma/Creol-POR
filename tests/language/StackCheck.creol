class Test
begin
  var depth : Int := 0
  var max : Int := 3

  op run == test(;)
  op test ==
    var temp: Int; depth := depth + 1; temp := depth;
    if depth <= max then test(;) end;
    await depth = temp; depth := depth - 1
end
