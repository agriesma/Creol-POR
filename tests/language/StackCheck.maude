load creol-interpreter
mod PROGRAM is
pr CREOL-INTERPRETER .
op init : -> Configuration [ctor] .
eq init =
< "Test" : Cl | Inh: noInh, Par: noVid, Att: "depth" |-> null , "max" |-> null, Mtds: 
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return ( emp ) > *
  < "run" : Mtdname | Param: noVid, Latt: "label:0" |-> null, Code: ( "label:0" ! "this" . "test" (emp ) )  ; ( "label:0" ? ( noVid ) )  ; return ( emp ) > *
  < "test" : Mtdname | Param: noVid, Latt: "label:0" |-> null , "temp" |-> null, Code: "depth" ::= "+" ( "depth" # int(1) ) ; "temp" ::= "depth" ; if "<=" ( "depth" # "max" ) th ( "label:0" ! "this" . "test" (emp ) )  ; ( "label:0" ? ( noVid ) )  el skip fi ; ( await "=" ( "depth" # "temp" ) ) ; "depth" ::= "-" ( "depth" # int(1) ) ; return ( emp ) >, Ocnt: 0 >

.
endm
