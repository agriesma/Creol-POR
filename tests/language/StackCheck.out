==========================================
reduce in PROGRAM : init .
result Class: 
< "Test" : Cl | Inh: noInh, Par: noVid, Att: "depth" |-> null, "max" |-> null,
    Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: (("depth" @@ "Test"),
    ("max" @@ "Test")) ::= int(0) # int(3) ; return(emp) > *
  < "run" : Mtdname | Param: noVid, Latt: "label:0" |-> null, Code: "label:0" !
    "this" . "test"(emp) ; "label:0" ?(noVid) ; return(emp) > *
  < "test" : Mtdname | Param: noVid, Latt: "label:0" |-> null, "temp" |-> null,
    Code: "depth" ::= "+"("depth" # int(1)) ; "temp" ::= "depth" ; if "<="(
    "depth" # "max") th "label:0" ! "this" . "test"(emp) ; "label:0" ?(noVid)
    el skip fi ; await "="("depth" # "temp") ; "depth" ::= "-"("depth" # int(
    1)) ; return(emp) >,
  Ocnt: 0 >
Bye.
